src/raw.rs

```diff
@@ -473,7 +473,7 @@ where
 {
     assert_eq!(fields.len(), values.len());
 
-    let mut fi = fields.iter();
+    let mut fi = fields.iter().map(|s| CString::new(s.clone()).unwrap());
```

Comment:
Are we sure there are no internal 0 in the data? Otherwise the unwrap here is wrong. We can either raise the error or (if we are sure) then maybe use `expect`?

Reply:
If the field names can contain internal NULLs then the entire use of `REDISMODULE_HASH_CFIELDS` is wrong. That would arguably be an implementation bug and not an error condition, so panicking is the correct way to handle it. (We can use an `expect` if we want to be nicer.)

A discussion on whether we should support non-string field names might be called for, but is outside the scope of this issue I think.

Reply:
I believe you need to collect the CString to another vector and then iterate over this vector by references (So you will get references to the CString's).

Reply:
Yeah I think you might be right. I was trying to handle it without extra allocations.